src/pages/index.js

```diff
@@ -0,0 +1,333 @@
+  cardNameEl.textContent = data.name;
+
+  const cardImageEl = cardElement.querySelector(".card__image");
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const deleteButton = cardElement.querySelector(".card__delete-btn");
+
+  function handleLike(evt, cardId) {
+    evt.preventDefault();
+    const likeButton = evt.target;
+    likeButton.disabled = true;
+    const isLiked = likeButton.classList.contains("card__like-btn_liked");
+
+    api
+      .changeLikeStatus(cardId, isLiked)
+      .then((res) => {
+        likeButton.classList.toggle("card__like-btn_liked");
+      })
+      .catch((err) => {
+        console.error("Error:", err);
+      })
+      .finally(() => {
+        likeButton.disabled = false;
+      });
+  }
+
+  cardLikeBtn.addEventListener("click", (evt) => {
+    handleLike(evt, data._id);
+  });
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+    previewModalCaptionEl.textContent = data.name;
+  });
+
+  deleteButton.addEventListener("click", (evt) => {
+    handleDeleteCard(cardElement, data._id);
+  });
+
+  return cardElement;
+}
+
+cancelButton.addEventListener("click", (evt) => {
+  closeModal(deleteModal);
+});
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const deleteButton = evt.submitter;
+  setButtonText({
+    btn: deleteButton,
+    isLoading: true,
+    defaultText: "Delete",
+    loadingText: "Deleting...",
+  });
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText({
+        btn: deleteButton,
+        isLoading: false,
+        defaultText: "Delete",
+        loadingText: "Deleting...",
+      });
+    });
+}
+
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+
+function handlePressEsc(evt) {
+  if (evt.key === "Escape") {
+    const modal = document.querySelector(".modal_opened");
+    if (modal) {
+      closeModal(modal);
+    }
+  }
+}
+
+function openModal(modal) {
+  document.addEventListener("keydown", handlePressEsc);
+  modal.classList.add("modal_opened");
+}
+
+function closeModal(modal) {
+  document.removeEventListener("keydown", handlePressEsc);
+  modal.classList.remove("modal_opened");
+}
+
+function renderCard(cardData, placement = "prepend") {
+  const cardElement = getCardElement(cardData);
+
+  if (placement === "append") {
+    cardsList.append(cardElement);
+  } else {
+    cardsList.prepend(cardElement);
+  }
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  const modal__submit = evt.submitter;
```

Comment:
`modal__submit`

Every variable in the project should be in `camelCase` 


`submitButton`